x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -212,20 +221,28 @@ void deleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete, TimeValu
         logger.info("starting snapshot retention deletion for [{}] snapshots", count);
         long startTime = nowNanoSupplier.getAsLong();
         int deleted = 0;
+        int failed = 0;
         for (Map.Entry<String, List<SnapshotInfo>> entry : snapshotsToDelete.entrySet()) {
             String repo = entry.getKey();
             List<SnapshotInfo> snapshots = entry.getValue();
             for (SnapshotInfo info : snapshots) {
-                deleteSnapshot(repo, info.snapshotId());
-                deleted++;
+                Optional<SnapshotHistoryItem> result = deleteSnapshot(repo, info);
```

Comment:
Hmm.. I think a listener-based approach might be better here, maybe something where the `deleteSnapshot` call is passed a function to execute on success and one for failure, these could then increment the `deleted` and `failed` counts (they'd have to be `AtomicInteger`s then) and it could index the history information also